src/baseline_rule.py

```diff
@@ -128,15 +145,102 @@ def get_port_array(self):
 
         return ports_array
 
+    def get_port_array_calico(self):
+        if not self.port_min:
+            return [], self.protocol
+        ports_array = []
+        for port in range(self.port_min, self.port_max + 1):
+            ports_array.append(port)
+
+        return ports_array, self.protocol
+
+    def _get_calico_policy_spec_second_direction(self, is_ingress):
+        policy_spec = {'types': ['Ingress']} if is_ingress else {'types': ['Egress']}
+        if is_ingress:
+            policy_selector = self._selectors_as_netpol_peer_calico(self.target)
+            rule_dict = {'action': 'Allow',
+                         'source': self._selectors_as_netpol_peer_calico(self.source)}
+        else:
+            policy_selector = self._selectors_as_netpol_peer_calico(self.source)
+            rule_dict = {'action': 'Allow',
+                         'destination': self._selectors_as_netpol_peer_calico(self.target)}
+
+        policy_spec.update(policy_selector)
+        rule_type = 'ingress' if is_ingress else 'egress'
+        policy_spec.update({rule_type: [rule_dict]})
+        return policy_spec
+
+    def _get_policy_type(self):
+        """
+        :return: a tuple of (1) is_ingress bool flag (2) str of policy type (Ingress/Egress)
+        :rtype (bool, str)
+        """
+        is_ingress = not isinstance(self.target, IpSelector)
+        return is_ingress, 'Ingress' if is_ingress else 'Egress'
+
+    @staticmethod
+    def _get_calico_policy_dict(policy_spec, policy_name):
+        return {
+            'apiVersion': 'projectcalico.org/v3',
+            'kind': 'GlobalNetworkPolicy',
+            'metadata': {'name': policy_name},
+            'spec': policy_spec
+        }
+
+    def to_global_netpol_calico(self):
+        """
+        Create Calico GlobalNetworkPolicy resources representing the connections specified by the rule.
+        Relevant for baseline rules with no namespace, thus using GlobalNetworkPolicy which applies to all namespaces.
+        Note that two GlobalNetworkPolicy resources may be required for allowing both directions of a connection.
+
+        :return: A Calico GlobalNetworkPolicy resources representing the connections specified by the rule
+        :rtype: (dict, union[dict, None])
+        """
+        is_ingress_policy, policy_type = self._get_policy_type()
+        policy_spec = dict()
+        policy_spec.update({'types': [policy_type]})
+        policy_selector = self._selectors_as_netpol_peer_calico(self.target if is_ingress_policy else self.source)
+
+        policy_spec.update(policy_selector)
+        ports_list, protocol = self.get_port_array_calico()
+        ports_dict = {'ports': ports_list}
+
+        rule_to_add = {'action': 'Allow'}
+        if protocol:
+            rule_to_add.update({'protocol': protocol})
+        if is_ingress_policy:
+            src_dict = self._selectors_as_netpol_peer_calico(self.source, False)
+            rule_to_add.update({'source': src_dict})
```

Comment:
```suggestion
            rule_to_add['source'] = src_dict
```
Avoid using `dict.update()` (to be more consistent with the rest of the code)

Reply:
done